back/app/services/surveys/results_generator.rb

```diff
@@ -1,27 +1,38 @@
 # frozen_string_literal: true
 
+# Base class for generating survey results - others inherit as follows:
+# - ResultsWithGroupGenerator - allow grouping of single results
+# - ResultsWithDateGenerator - allow filtering by year and quarter
+# - ResultsWithLogicGenerator - apply survey logic to the results
+
 module Surveys
   class ResultsGenerator < FieldVisitorService
-    def initialize(phase)
+    def initialize(phase, structure_by_category: false)
       super()
+      @phase = phase
       form = phase.custom_form || CustomForm.new(participation_context: phase)
-      @fields = IdeaCustomFieldsService.new(form).enabled_fields
-      @inputs = phase.ideas.supports_survey.published
```

Comment:
We done seem to use this "enabled_fields" anymore in these lines of code - not sure what it does, but just want to double-check that's correct?